filter_plugins/host_addresses.py

```diff
@@ -0,0 +1,34 @@
+"""
```

Comment:
I don't think we need this filter.  We're using this to extract the values of `ansible_host` for a group of hosts, but we can get the same thing using built-in filters like this:

    rabbit_hosts: "{{ groups['rabbit_hosts']|map('extract', hostvars, ['ansible_host'])|list }}"

In fact, this seems to work better than your current code, because if I look at the value of `rabbit_hosts` that results from this `host_addresses` filter, it's not actually a list.  Given this playbook:

```
- hosts: all

- hosts: localhost
  tasks:
    - set_fact:
        my_hosts: >
          {{ groups['rabbit_hosts']|map('extract', hostvars, ['ansible_host'])|list }}
    - debug:
        var: my_hosts
    - debug:
        var: rabbit_hosts

```
Run as:

    ansible-playbook -i inventory hostaddrs.yml

I get as output:

```
TASK [debug] *******************************************************************
ok: [localhost] => {
    "my_hosts": [
        "192.0.2.15"
    ]
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "rabbit_hosts": "[u'192.0.2.15']\"\n"
}

```

See the difference?